apps/modernization-ui/src/components/button/NBSButton.tsx

```diff
@@ -0,0 +1,63 @@
+import { Button } from '@trussworks/react-uswds';
+import { ReactNode } from 'react';
+import styles from './NBSButton.module.scss';
+
+type Props = {
+    icon?: ReactNode;
+    label?: string;
+    outline?: boolean;
+    destructive?: boolean;
+    disabled?: boolean;
+    type?: 'button' | 'submit' | 'reset';
+    unstyled?: boolean;
+    labelPosition?: 'left' | 'right';
+    onClick?: () => void;
+} & JSX.IntrinsicElements['button'];
+
+const NBSButton = ({
+    type = 'button',
+    icon,
+    label,
+    outline = false,
+    destructive = false,
+    unstyled = false,
+    labelPosition = 'left',
+    ...defaultProps
+}: Props) => {
+    const isIconOnly = icon && !label;
+    const classesArray = [];
+    if (destructive) {
+        classesArray.push(styles.destructive);
+    }
+
+    if (isIconOnly) {
+        classesArray.push(styles.icon);
+    }
```

Comment:
suggestion (non-blocking):  The [classNames](https://github.com/JedWatson/classnames#readme) function can be used to handle conditionally applying classes to a component.  An object can be passed to the function with class names that are applied if the property value is `true`.

When using SASS modules bracket notation has to be used.  

```typescript

classNames({
  [styles.destructive] : styles.destructive,
  [styles.icon] : styles.icon
});

```